loopd/daemon.go

```diff
@@ -242,6 +208,10 @@ func (d *Daemon) Run() error {
 	select {
 	case <-interruptChannel:
 		log.Infof("Received SIGINT (Ctrl+C).")
+		err := d.stop()
```

Comment:
With the prior `defer` calls, the cleanup was always performed (also for errors). Now it is only when an interrupt happens?

Reply:
I added a `Stop` to each error case. So a proper shutdown should now always be performed, not only on interrupt.

Reply:
Isn't this now already cleaning up dependencies before the main `cancel()` is called?

Reply:
Discussed offline. The stop now first stops accepting gRPC connections, then cancels the main context, shuts down the gRPC server and then its dependencies.